addons/LPCAnimatedSprite/LPCAnimatedSprite2D.gd

```diff
@@ -34,13 +34,89 @@ enum LPCAnimation {
 	IDLE_RIGHT,
 	HURT_DOWN_LAST
 }
+
+#### NO LONGER NEEDED (WITH CURRENT APPROCH), FOR REFERENCE ONLY ####
+var anim_lookup_dict: Dictionary = {
+	"IDLE": {
+		Vector2(0,-1): ["IDLE_UP", "OVERSIZE_IDLE_UP"],
+		Vector2(-1,0): ["IDLE_LEFT", "OVERSIZE_IDLE_LEFT"],
+		Vector2(0,1): ["IDLE_DOWN", "OVERSIZE_IDLE_DOWN"],
+		Vector2(1,0): ["IDLE_RIGHT", "OVERSIZE_IDLE_RIGHT"],
+	},
+	"WALK": {
+		Vector2(0,-1): ["WALK_UP", "OVERSIZE_WALK_UP"],
+		Vector2(-1,0): ["WALK_LEFT", "OVERSIZE_WALK_LEFT"],
+		Vector2(0,1): ["WALK_DOWN", "OVERSIZE_WALK_DOWN"],
+		Vector2(1,0): ["WALK_RIGHT", "OVERSIZE_WALK_RIGHT"],
+	},
+	"THRUST": {
+		Vector2(0,-1): ["THRUST_UP", "OVERSIZE_THRUST_UP"],
+		Vector2(-1,0): ["THRUST_LEFT", "OVERSIZE_THRUST_LEFT"],
+		Vector2(0,1): ["THRUST_DOWN", "OVERSIZE_THRUST_DOWN"],
+		Vector2(1,0): ["THRUST_RIGHT", "OVERSIZE_THRUST_RIGHT"],
+	},
+	"CAST": {
+		Vector2(0,-1): ["CAST_UP", "OVERSIZE_CAST_UP"],
+		Vector2(-1,0): ["CAST_LEFT", "OVERSIZE_CAST_LEFT"],
+		Vector2(0,1): ["CAST_DOWN", "OVERSIZE_CAST_DOWN"],
+		Vector2(1,0): ["CAST_RIGHT", "OVERSIZE_CAST_RIGHT"],
+	},
+	"SLASH": {
+		Vector2(0,-1): ["SLASH_UP", "OVERSIZE_SLASH_UP"],
+		Vector2(-1,0): ["SLASH_LEFT", "OVERSIZE_SLASH_LEFT"],
+		Vector2(0,1): ["SLASH_DOWN", "OVERSIZE_SLASH_DOWN"],
+		Vector2(1,0): ["SLASH_RIGHT", "OVERSIZE_SLASH_RIGHT"],
+	},
+	"SHOOT": {
+		Vector2(0,-1): ["SHOOT_UP", "OVERSIZE_SHOOT_UP"],
+		Vector2(-1,0): ["SHOOT_LEFT", "OVERSIZE_SHOOT_UP"],
+		Vector2(0,1): ["SHOOT_DOWN", "OVERSIZE_SHOOT_DOWN"],
+		Vector2(1,0): ["SHOOT_RIGHT", "OVERSIZE_SHOOT_RIGHT"],
+	},
+}
+
+var cur_dir: Vector2
 var AnimationNames:Array
+var reference_sheet := LPCSpriteSheet.new().NormalAnimationData
```

Comment:
Just ocurred to me that all animation data should be static var

Reply:
> Just ocurred to me that all animation data should be static var, I'm changing it on my code

happy to help 😛